WordPress/Classes/ViewRelated/System/Search/WPSearchControllerConfigurator.m

```diff
@@ -0,0 +1,112 @@
+#import "WPSearchControllerConfigurator.h"
+#import "WordPress-Swift.h"
+
+const CGFloat SearchBarWidth = 280.0;
+const CGFloat SearchBariPadWidth = 600.0;
+const CGFloat SearchWrapperViewPortraitHeight = 64.0;
+const CGFloat SearchWrapperViewLandscapeHeight = 44.0;
+const NSTimeInterval SearchBarAnimationDuration = 0.2; // seconds
+
+@interface WPSearchControllerConfigurator ()
+
+@property (nonatomic, weak) WPSearchController *searchController;
+@property (nonatomic, weak) UIView *searchWrapperView;
+@property (nonatomic, weak) UISearchBar *searchBar;
+
+@end
+
+@implementation WPSearchControllerConfigurator
+
+- (instancetype)initWithSearchController:(WPSearchController *)searchController
+                   withSearchWrapperView:(UIView *)searchWrapperView
+{
+    self = [super init];
+    if (self) {
+        _searchController = searchController;
+        _searchWrapperView = searchWrapperView;
+        _searchBar = _searchController.searchBar;
+    }
+    
+    return self;
+}
+
+- (void)configureSearchControllerBarAndWrapperViewOfClass:(Class)class
+{
+    [self configureSearchController];
+    [self configureSearchBar:class];
+    [self configureSearchWrapper];
+}
+
+- (void)configureSearchController
+{
+    self.searchController.dimsBackgroundDuringPresentation = NO;
+    self.searchController.hidesNavigationBarDuringPresentation = YES;
+}
+
+- (void)configureSearchBar:(Class)class
+{
+    [self configureSearchBarPlaceholder:class];
+    
+    self.searchBar.translatesAutoresizingMaskIntoConstraints = NO;
+    self.searchBar.accessibilityIdentifier = @"Search";
+    self.searchBar.autocapitalizationType = UITextAutocapitalizationTypeNone;
+    self.searchBar.backgroundImage = [[UIImage alloc] init];
+    self.searchBar.tintColor = [WPStyleGuide grey]; // cursor color
+    self.searchBar.translucent = NO;
+    self.searchBar.barStyle = UIBarStyleBlack;
+    self.searchBar.barTintColor = [WPStyleGuide wordPressBlue];
+    self.searchBar.showsCancelButton = YES;
+    [self.searchBar setImage:[UIImage imageNamed:@"icon-clear-textfield"] forSearchBarIcon:UISearchBarIconClear state:UIControlStateNormal];
+    [self.searchBar setImage:[UIImage imageNamed:@"icon-post-list-search"] forSearchBarIcon:UISearchBarIconSearch state:UIControlStateNormal];
+    
+    [self configureSearchBarForSearchView:class];
+}
+
+- (void)configureSearchBarPlaceholder:(Class)class
+{
+    // Adjust color depending on where the search bar is being presented.
+    UIColor *placeholderColor = [WPStyleGuide wordPressBlue];
+    NSString *placeholderText = NSLocalizedString(@"Search", @"Placeholder text for the search bar on the post screen.");
+    NSAttributedString *attrPlacholderText = [[NSAttributedString alloc] initWithString:placeholderText attributes:[WPStyleGuide defaultSearchBarTextAttributes:placeholderColor]];
+    [[UITextField appearanceWhenContainedIn:[UISearchBar class], class, nil] setAttributedPlaceholder:attrPlacholderText];
```

Comment:
@aerych That's a good I believe. I was so focused on reducing redundancy that I passed in everything necessary. That's actually where the whole `id<WPSearchControllerDelegate, WPSearchResultsUpdating>` parameter came up. I'll remove the class parameter and have the VCs do it.

@diegoreymendez I appreciate that opinion! It's good to get perspective of someone seeing it with less context. Do you have any suggestions on how to make it less obscure? Or is a more descriptive name/documentation good enough you think?

> the fact that two parent VCs set up a child view in the same way does not mean it's bound to stay like that, and may be purely coincidental for a given period of time.

Yes, you're right. However this instance benefits from a central place where the work is done I believe.